core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java

```diff
@@ -0,0 +1,31 @@
+package com.baeldung.streams.minmaxbygroup;
+
+import java.util.DoubleSummaryStatistics;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+/** The type Order processor. */
+public class OrderProcessor {
+  /**
+   * Group by category with min max per group.
+   *
+   * @param orderItems the order items
+   * @return the grouped items
+   */
+  public Map<OrderItemCategory, Pair<Double, Double>> groupByCategoryWithMinMax(
+      List<OrderItem> orderItems) {
+    Map<OrderItemCategory, DoubleSummaryStatistics> categoryStatistics =
+        orderItems.stream()
```

Comment:
you can keep the orderItems.stream() in the same line as variable declaration

Reply:
I had to format that manually. I was using `intellij-baeldung-style.xml`, then the formatting takes it to next line.